test/unit/specs/utils/countDownLatch.spec.js

```diff
@@ -0,0 +1,32 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { countDownLatch } from "../../../../src/utils";
+
+describe("countDownLatch", () => {
+  it("should unlock after all count downs", () => {
+    const latch = countDownLatch(2);
+    const spy = jasmine.createSpy();
+
+    setTimeout(() => {
+      latch.countDown();
+      latch.countDown();
+    }, 0);
+
+    return latch
```

Comment:
I thought about this too, but I think there might be a blind spot with that approach. If `wait` were to return `undefined`, for example, I believe Jasmine would assume success. The way @artur-ciocanu wrote the test ensures that `wait` returns a promise. It might be reasonable enough to assume `wait` returns a promise though. Anyway, I don't know if that's why @artur-ciocanu did it this way, but it might deserve a comment if both @jonsnyder and I thought about removing the stuff after `wait`, because some later developer might do the same.

Reply:
You could assert to see if `latch.wait()` returns a promise before returning it though. 🤷‍♂